internal/aws/client/client.go

```diff
@@ -0,0 +1,65 @@
+package client
+
+import (
+	"context"
+	"github.com/Falco-Talon/falco-talon/utils"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/lambda"
+)
+
+type AWSClient struct {
+	cfg aws.Config
+
+	mu           sync.Mutex // Protects the fields below
+	lambdaClient *lambda.Client
+	//s3Client     *s3.Client
+}
+
+var (
+	awsClient *AWSClient
+	once      sync.Once
+)
+
+func Init() error {
+	var err error
+	once.Do(func() {
```

Comment:
I didn't know that way to avoid multi inits, do you think we could use it for kubernetes client too?

Reply:
We certainly can. I need to test, but based on the current implementation, we initialize the client whenever rules are changed. We need to understand if this is the behavior we want or not. Do you plan to allow customers to specify client configs via rule/action definition? 

For instance configuring client to disable SSL validation, in that case we would need to always initialize it with new configs so the once.Do() won't do the work properly. 

WDYT?

Reply:
The rules are dynamic (can be hot reloaded) but config for the clients is static. After some thoughts, I think the client should be init only once. The first time a rule needing it is encountered.